crates/host/src/wasmbus/mod.rs

```diff
@@ -369,6 +369,40 @@ fn linkdef_hash(
     hex::encode_upper(hash.finalize())
 }
 
+#[instrument(skip(jetstream))]
+async fn create_lattice_metadata_bucket(
+    jetstream: &context::Context,
+    bucket: &str,
+) -> anyhow::Result<()> {
+    match jetstream
+        .create_key_value(kv::Config {
+            bucket: bucket.to_string(),
+            ..Default::default()
+        })
+        .await
+    {
+        Ok(_) => {
+            info!("created lattice metadata bucket {bucket} with 1 replica");
+            Ok(())
+        }
+        Err(err) => {
+            // Ignore the error if the bucket was pre-provisioned
+            // This is a terrible hack, but the error we get from the NATS client is a nested
+            // dyn Error. The NATS server will return an error containing the code
+            // 10058 if the underlying stream already exists with different configuration,
+            // which is the case we care about
+            if err.to_string().contains("10058") {
```

Comment:
Can't we just downcast?

Reply:
Looking at the docs, you need to call https://docs.rs/async-nats/latest/async_nats/jetstream/context/struct.CreateKeyValueError.html#method.kind first to find out the kind of the error and then use https://docs.rs/async-nats/latest/async_nats/jetstream/context/struct.CreateKeyValueError.html#method.source to acquire the internal error, downcast it to https://docs.rs/async-nats/latest/async_nats/jetstream/struct.Error.html and check the code

Reply:
I looked into this before opening the PR and talked it through with @thomastaylor312

Rather than chain those functions to extract the error, we decided to just flatten the error hierarchy to a string and search for the error code

Reply:
Why? This clearly not the right approach, what's the motivation behind this?

Reply:
Because while flattening and searching for a substring is brittle, so is chaining and downcasting, and the latter is much harder to read

Reply:
This is a very subjective comment and I disagree about the "brittleness" of downcasting and checking the exact error code, please expand on why exactly you think it's brittle?

Reply:
I walked through this with @connorsmith256 and I agree with the shortcut here. We have to downcast at least 2 (though I think it was 3) times, which involves a lot of option handling and is just a pain to read. Given that this was for a _single_ use case at the beginning of startup, this felt like a great corner to cut and was more readable. These NATS error codes are pretty stable so I don't worry about "cheating" here